### ISSUE TITLE
save() and related models

### ISSUE THREAD

**Issue by leth** [creatoro/jelly#89]
I came across the fact that save() does not ensure that related models are also saved (updated or inserted).

Is this by design, or merely an oversight?

Circular relationships have the potential to cause loops if this isn't implemented carefully.

**Comment by loonies** [comment]
Do you mean saving entire object graph?

e.g.

```
$category = Jelly::factory('Category');
$category->name = 'foo';  // category foo does not exist yet

$news->title = 'bar';
$news->category = $category;

$news->save();
```

Now, we should have 2 saved objects, category and news, with one `save()` call.

**Comment by leth** [comment]
Yeah. I think this needs some proper research as to whether it's a good thing to do or not.

**Comment by loonies** [comment]
That would be an awesome feature to have. Some problems:
- dependency resolution (category must be saved before news)
- collecting error messages

**Comment by creatoro** [comment]
Sounds as a cool and problematic feature :).

**Comment by paulchubatyy** [comment]
There also might be a reason to use transactions for storing the object with related objects and reverting the whole thing if one of objects does not validate. This is really a complex task

**Comment by ccazette** [comment]
Too much magic IMHO... Don't make it bloated !
I am not saying such features wouldn't be a good thing but keeping the core clean and lean should be the priority.. May be it is time to consider a modular approach of Jelly (for instance I can say I could really do without the whole behavior / event system).. Some modules could bring additional features (such as this, polymorphism or transaction support) to a faster and cleaner core...
Just my 2 cents...

**Comment by creatoro** [comment]
@ccazette: lately I've been thinking about modularizing Jelly as well... Could worth the hassle.

However, the most important would be to have tests for **everything**.
